### PR TITLE
Adds Filamentphp documentation search extension

### PR DESCRIPTION
**Pull request by aurawindsurfing**
Adds filamentphp documentation search extension

## Description

Adds search extension for https://filamentphp.com/ admin panel documentation

## Screencast

Screenshot 2022-07-02 at 14 03 56

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @aurawindsurfing 👋 

Thanks for another contribution 💪 

- Should we consider just calling it Filament instead?

- Is it possible to have some initial items on the list so it's not all empty?

- You could consider a [List.EmptyView](https://developers.raycast.com/api-reference/user-interface/list#list.emptyview) to make the UI even better (with the logo in 64x64)

- When searching for `test` it gives me this result, it's not really saying anything useful, could you look into this?

- Add [metadata images](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots)

CleanShot 2022-07-06 at 13 36 23@2x

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by aurawindsurfing** [comment]
I have no idea why exactly same code in both extensions throws error here:

`<List.EmptyView icon="empty-state-icon2.png" title="Whoops! We did not find any results!"/>`

I would appreciate if you @pernielsentikaer  can have a quick look at it as I'm puzzled by it.

**Comment by zepfietje** [comment]
> > * Should we consider just calling it Filament instead?
> 
> I think it should stay as "filamentphp" the name is used interchangebly with "filament" but this gives is more reference to the actual package

I think the name should actually be changed to Filament. Let's ask the creator himself, @danharrin.

**Comment by danharrin** [comment]
Yes, `Filament` is great

File: extensions/filamentphp-doc-seach/src/filament.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ export default function SearchFilamentphpDocumentation() {
         return Object.entries(_.groupBy(res.hits, "hierarchy.lvl1")) || [];
       })
       .catch((err) => {
-        showToast(Toast.Style.Failure, "Error searching filamentphp docs", err.message);
+        showToast(Toast.Style.Failure, "Error searching filament documentation", err.message);
```
Comment (zepfietje) [inline extensions/filamentphp-doc-seach/src/filament.tsx:60]: Shouldn't this be `Error searching Filament documentation`?

File: extensions/filamentphp-doc-seach/package.json [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
   "license": "MIT",
   "commands": [
     {
-      "name": "filamentphp",
+      "name": "filament",
       "title": "Search Documentation",
-      "subtitle": "Filamentphp",
-      "description": "Searches the filamentphp documentation",
+      "subtitle": "Filament",
+      "description": "Searches the filament documentation",
```
Comment (zepfietje) [inline extensions/filamentphp-doc-seach/package.json:14]: `filament -> Filament`

**Comment by zepfietje** [comment]
> @pernielsentikaer done!

@aurawindsurfing, could you have a look at my minor tweaks?

File: extensions/filamentphp-doc-seach/README.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
-# Filamentphp Admin Documentation Search
+# Filament Admin Documentation Search
```
[the remark this answers is not in the document]
Comment (pernielsentikaer) [reply extensions/filamentphp-doc-seach/README.md:1]: Good catch, I also found it last time confusing but forgot about it

**Comment by pernielsentikaer** [comment]
I think I done the rest, could you take a last look @zepfietje 🙂

**Comment by aurawindsurfing** [comment]
> I think I done the rest, could you take a last look @zepfietje 🙂

Thanks @pernielsentikaer

**Comment by aurawindsurfing** [comment]
@pernielsentikaer what is the accepted practice then for not commiting node_modules to your repo without commiting .gitignore?

**Comment by pernielsentikaer** [comment]
> @pernielsentikaer what is the accepted practice then for not committing node_modules to your repo without committing .gitignore?

We have [this file](https://github.com/raycast/extensions/blob/main/.gitignore) in the root of the repository (which should cover `node_modules` and other regular cases)

**Comment by aurawindsurfing** [comment]
> I'm not sure if it was you @aurawindsurfing, but I can see that we both have the old folder structure and a new one, do you know how to clean (and removing the wrong one)

I'm sorry but no. I did not notice any differences

**Comment by zepfietje** [comment]
> I think I done the rest, could you take a last look @zepfietje 🙂

Sure, @pernielsentikaer! Let me do a quick review.

File: extensions/filamentphp-doc-seach/src/filament.tsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+      setSearchResults(await search());
+      setIsLoading(false);
+    })();
+  }, []);
+
+  return (
+    <List
+      throttle={true}
+      isLoading={isLoading}
+      searchBarPlaceholder={"Search Filament"}
```
Comment (zepfietje) [inline extensions/filamentphp-doc-seach/src/filament.tsx:76]: In some other extensions the placeholder is just `Search...`

File: extensions/filamentphp/README.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# Filamentphp Admin Documentation Search
```
Comment (zepfietje) [inline extensions/filamentphp/README.md:1]:
```suggestion
# Filament
```

File: extensions/filamentphp/README.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+
+![Raycast Filamntphp Search](media/filamentphp.gif)
```
Comment (zepfietje) [inline extensions/filamentphp/README.md:3]:
```suggestion
![Raycast Filament extension](media/filamentphp.gif)
```

File: extensions/filamentphp/README.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
...
+
+[Filamentphp](https://filamentphp.com) search
```
Comment (zepfietje) [inline extensions/filamentphp/README.md:5]:
```suggestion
[Filament](https://filamentphp.com)
```

File: extensions/filamentphp-doc-seach/package.json [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  "description": "Documentation Search",
+  "icon": "command-icon2.png",
+  "author": "aurawindsurfing",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "filament",
+      "title": "Search Documentation",
+      "subtitle": "Filament",
+      "description": "Searches the Filament documentation",
```
Comment (zepfietje) [inline extensions/filamentphp-doc-seach/package.json:14]:
```suggestion
      "description": "Search the Filament documentation",
```

**Review by zepfietje — COMMENTED** [review]
General remarks:

There are 2 folders: filamentphp-doc-search and filamentphp. Shouldn't this just be 1 named filament?

Also, the images are still named filamentphp-x.xxx which should also be changed to just filament.

Comparing to other extensions like Tailwind CSS and Laravel Livewire, I added some comments in this review regarding description and placeholder text. Not too sure what the Raycast guidelines are, but I tried to at least align it between extensions.

**Comment by aurawindsurfing** [comment]
@zepfietje have a look here also once we are at the Laravel stuff 🚀

https://github.com/raycast/extensions/pull/2190

**Comment by zepfietje** [comment]
> General remarks:
> 
> There are 2 folders: filamentphp-doc-search and filamentphp. Shouldn't this just be 1 named filament?
> 
> Also, the images are still named filamentphp-x.xxx which should also be changed to just filament.
> 
> Comparing to other extensions like Tailwind CSS and Laravel Livewire, I added some comments in this review regarding description and placeholder text. Not too sure what the Raycast guidelines are, but I tried to at least align it between extensions.

@pernielsentikaer @aurawindsurfing, could you still address these remarks?

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Comment by zepfietje** [comment]
Seems like the entire extension folder is now named `filamentphp-doc-seach` instead of `filament`? (Even a typo in `seach -> search`). @pernielsentikaer

**Comment by aurawindsurfing** [comment]
> Seems like the entire extension folder is now named `filamentphp-doc-seach` instead of `filament`? (Even a typo in `seach -> search`). @pernielsentikaer

I will leave it as it is for @pernielsentikaer do not want to do another PR, not sure of the procedure of already published extensions

**Comment by pernielsentikaer** [comment]
It's something that we can handle but I have to change something behind the curtains before doing it, it's not really changing  anything in the end since end users won't even see it - but I'll add it to my list :)

**Pull request merged**